File: cutils.platform/src/Parity.cpp
#include "Parity.h"

using std::string;

namespace CUtils
{

/*******************************************************************************
 *
 ******************************************************************************/
string toString(Parity p)
{
    switch (p)
    {
    case Parity::NONE:
        return "None";

    case Parity::ODD:
        return "Odd";

    case Parity::EVEN:
        return "Even";

    case Parity::MARK:
        return "Mark";

    case Parity::SPACE:
        return "Space";
    }

    return "Undefined";
}

} // namespace CUtils

File: jutils.platform/src/jutils/platform/data/Parity.java
package jutils.platform.data;

import jutils.core.INamedValue;

/*******************************************************************************
 * 
 ******************************************************************************/
public enum Parity implements INamedValue
{
    /**  */
    NONE( 0, "None" ),
    /**  */
    ODD( 1, "Odd" ),
    /**  */
    EVEN( 2, "Even" ),
    /**  */
    MARK( 3, "Mark" ),
    /**  */
    SPACE( 4, "Space" );

    /**  */
    public final int value;
    /**  */
    public final String name;

    /***************************************************************************
     * @param value
     * @param name
     **************************************************************************/
    private Parity( int value, String name )
    {
        this.value = value;
        this.name = name;
    }

    /***************************************************************************
     * {@inheritDoc}
     **************************************************************************/
    @Override
    public String getName()
    {
        return name;
    }

    /***************************************************************************
     * {@inheritDoc}
     **************************************************************************/
    @Override
    public int getValue()
    {
        return value;
    }
}
